exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/metadata/Metadata.java

```diff
@@ -633,43 +716,169 @@ private void readBlockMeta(Path path, boolean dirsOnly, MetadataContext metaCont
+    if (metadataSummary == null) {
+      return null;
+    } else {
+      Set<String> interestingColumns = new HashSet<String>();
+      for (ColumnTypeMetadata_v4 columnTypeMetadata_v4: metadataSummary.columnTypeInfo.values()) {
+        if (columnTypeMetadata_v4.isInteresting) {
+          interestingColumns.add(String.join("", columnTypeMetadata_v4.name));
+        }
+      }
+      return interestingColumns;
+    }
+  }
+
+  private boolean getallColumnsInteresting(FileSystem fs, Path metadataParentDir, boolean autoRefreshTriggered) {
+    Metadata_V4.MetadataSummary metadataSummary = getSummary(fs, metadataParentDir, autoRefreshTriggered, null);
+    if (metadataSummary == null) {
+      return true;
+    }
+    return metadataSummary.isAllColumnsInteresting();
+  }
+
+  private static Path getSummaryFileName(Path metadataParentDir) {
+    Path summaryFile = new Path(metadataParentDir, METADATA_SUMMARY_FILENAME);
+    return summaryFile;
+  }
+
+  private static Path getDirFileName(Path metadataParentDir) {
+    Path metadataDirFile = new Path(metadataParentDir, METADATA_DIRECTORIES_FILENAME);
+    return metadataDirFile;
+  }
+
+  private static Path getFileMetadataFileName(Path metadataParentDir) {
+    Path fileMetadataFile = new Path(metadataParentDir, METADATA_FILENAME);
+    return fileMetadataFile;
+  }
+
+  /**
+   * Returns if metadata exists or not in that directory
+   * @param fs filesystem
+   * @param metadataParentDir parent directory that holds metadata files.
+   * @return true if metadata exists in that directory
+   * @throws IOException
+   */
+  private static boolean metadataExists(FileSystem fs, Path metadataParentDir) throws IOException {
+    Path summaryFile = new Path(metadataParentDir, METADATA_SUMMARY_FILENAME);
+    Path metadataDirFile = new Path(metadataParentDir, METADATA_DIRECTORIES_FILENAME);
+    Path fileMetadataFile = new Path(metadataParentDir, METADATA_FILENAME);
+    if (!fs.exists(summaryFile) && !fs.exists(metadataDirFile) && !fs.exists(fileMetadataFile)) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Reads the summary from the metadata cache file, if the cache file is stale recreates the metadata
+   * @param fs
+   * @param metadataParentDir
+   * @param autoRefreshTriggered true if the auto-refresh is already triggered
+   * @param readerConfig
+   * @return returns metadata summary
+   */
+  public static Metadata_V4.MetadataSummary getSummary(FileSystem fs, Path metadataParentDir, boolean autoRefreshTriggered, ParquetReaderConfig readerConfig) {
+    Path summaryFile = getSummaryFileName(metadataParentDir);
+    Path metadataDirFile = getDirFileName(metadataParentDir);
+    MetadataContext metaContext = new MetadataContext();
+    try {
+      // If autoRefresh is not triggered and none of the metadata files exist
+      if (!autoRefreshTriggered && metadataExists(fs, metadataParentDir)) {
```

Comment:
This condition check does not look right..shouldn't it check metadataDirFile ? and should it be based on autoRefreshTriggered being true ?

Reply:
So this is the case where getSummary is called to get the row count and null count while applying convertcounttodirectscan. This check looks if any of the metadata cache files exist at all in the directory. If none of the cache files are found implies that metadata doesn't exist for that directory so null is returned in that case. The autorefreshtriggered is set to true only while getting interesting columns from the previous summary file.

Reply:
hmm..so shouldn't this be `! metadataExists()` ?

Reply:
Yes, Changed it.